doc/advanced-testing-mini-course/src/ATMC/Lec5/StateMachineDSL.hs

```diff
@@ -18,15 +18,15 @@ import ATMC.Lec5.Time
 
 ------------------------------------------------------------------------
 
-type SMM s msg resp a = StateT s (StateT StdGen (Writer [Output resp msg])) a
+type SMM s msg resp a = StateT s (StateT StdGen (WriterT [Output resp msg] Maybe)) a
 
 newtype Seed = Seed Int
 
 runSMM :: Seed -> SMM s msg resp () -> s -> ([Output resp msg], s)
-runSMM (Seed seed) m s = (outputs, s')
-  where
-    ((((), s'), _stdGen'), outputs) =
-      runWriter (runStateT (runStateT m s) (Random.mkStdGen seed))
+runSMM (Seed seed) m s =
+  case runWriterT (runStateT (runStateT m s) (Random.mkStdGen seed)) of
+    Nothing                              -> ([], s)
+    Just ((((), s'), _stdGen'), outputs) -> (outputs, s')
```

Comment:
So unrelated to this change, but doesn't this mean that if I'm using random, we will not propagate the new seed? It seems we didn't do it before this commit so don't change it in this PR.